.github/workflows/ci.yml

```diff
@@ -6,6 +6,9 @@ on:
     # https://crontab.guru/every-3-hours
     - cron: "0 */3 * * *"
   workflow_dispatch:  # allows manual triggering with branch picker
+  push:
+    tags:
+      - '[0-9]+.[0-9]+.[0-9]+'
```

Comment:
Nit - It would be good to have a top-level docstring explaining the different cases that this CI is run. Not necessary for this PR but would be a nice way to wrap up the effort imo

Reply:
Although I guess we have some good comments above - will leave it up to you and the rest of the team